### PR TITLE
Containerize NTT (attempt 2)

### PR DESCRIPTION
**Pull request by mathcolo**
This reverts commit 7ff758fe451db70df355a2cbe503e86fc74a1f5c.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      postgres:
+        condition: service_healthy
+    build:
+      context: .
+    restart: on-failure
+    environment:
+      - NEW_BUILD=true
+      - POSTGRES_HOST=postgres
+    ports:
+      - "5001:5001"
```
Comment (thiteixeira) [inline docker-compose.yml:15]: I would be nice to expose a volume mount here. It's something I haven't done yet with Docker compose though. Not sure if it has to be done here or in the Dockerfile, but the idea is to re-compile if there are changes in the local repo

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      - APP_DB_PASS=ubuntu
+      - APP_DB_NAME=newtrains
+    volumes:
+      - ./db:/docker-entrypoint-initdb.d/
+    expose:
+      - "5432"
+
+networks:
+  default:
+    name: transit
```
Comment (thiteixeira) [inline docker-compose.yml:38]: Nit: add line at the end of file

File: package.json [hunk 3]
```diff
@@ -32,6 +32,7 @@
   "dependencies": {
     "@types/react": "^18.0.1",
     "@types/react-dom": "^18.0.0",
+    "acorn": "^8.8.0",
```
Comment (thiteixeira) [inline package.json:35]: Should react and react-dom match exactly?

**Review by thiteixeira — APPROVED** [review]
LGTM

**Comment by nathan-weinberg** [comment]
Once we have this working, we could update the CD to update the repo on the Lightsail instance, rebuild the image from the newest Dockerfile, and restart the container?

File: wait-for-it.sh [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+until ping -c 1 postgres >/dev/null 2>&1; do echo "not up"; done
```
Comment (nathan-weinberg) [inline wait-for-it.sh:2]:
```suggestion
#!/bin/sh

until ping -c 1 postgres >/dev/null 2>&1; do echo "not up"; done
```

**Comment by nathan-weinberg** [comment]
Could we add README details as to how to work everything?

**Comment by mathcolo** [comment]
Yep! This is blocked on what to do about `psycopg2` (switching from `psycopg2-binary` to `psycopg2` broke the current instance, even though the Docker version needed it). After that's fixed we should be ready for documentation.

Maybe we should hold off on merging this until we're ready on the Ansible/CloudFormation side for containerized deployment, though? The mainline is not broken right now, so it might be prudent not to risk breaking it again.

**Pull request closed without merging**